restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -1251,16 +1251,16 @@ private BiFunction<Optional<JsonPointer>, JsonNode, String> documentIdResolver()
             .map(
                 p -> {
                   JsonNode node = jsonNode.at(p);
-                  if (!node.isTextual()) {
+                  if (!node.isValueNode() || node.isMissingNode()) {
```

Comment:
`|| node.isMissingNode()` might not be needed because `.isValueNode()` include that check already.

```java
    public final boolean isValueNode()
    {
        switch (getNodeType()) {
            case ARRAY: case OBJECT: case MISSING:
```

Reply:
I wasn't sure - thanks!

Reply:
Is `null` a valid scalar for this case? In other words should a `null` scalar be an error or is that okay?

Reply:
Let's avoid `null` that makes no sense..